relation_engine/batchload/delta_load.py

```diff
@@ -47,68 +56,133 @@ def load_graph_delta(
          specified.
     """
     db = database
+    if merge_source and not db.get_merge_collection():
+        raise ValueError('A merge source is specified but the database ' +
+           'has no merge collection')
+    _process_verts(db, vertex_source, timestamp, load_version)
+    if merge_source:
+        _process_merges(db, merge_source, timestamp, load_version)
     
-    # count = 0
-    for v in vertex_source:
-        # could batch things up here if slow
-        dbv = db.get_vertices([v[_ID]], timestamp).get(v[_ID])
-        if not dbv:
-            db.save_vertex(v[_ID], load_version, timestamp, v)
-        elif not _special_equal(v, dbv):
-            db.expire_vertex(dbv[_KEY], timestamp - 1)
-            db.save_vertex(v[_ID], load_version, timestamp, v)
-        else:
-            # mark node as seen in this version
-            db.set_last_version_on_vertex(dbv[_KEY], load_version)
-        # count += 1
-        # if count % 1000 == 0:
-        #     print(f'node {count}')
+    if _VERBOSE: print(f'expiring vertices: {time.time()}')
+    db.expire_extant_vertices_without_last_version(timestamp - 1, load_version)
 
-    if merge_source:
-        for m in merge_source:
-            dbmerged = db.get_vertices([m['from']], timestamp).get(m['from'])
-            dbtarget = db.get_vertices([m['to']], timestamp).get(m['to'])
+    _process_edges(db, edge_source, timestamp, load_version)
+    
+    if _VERBOSE: print(f'expiring edges: {time.time()}')
+    for col in db.get_edge_collections():
+        db.expire_extant_edges_without_last_version(
+            timestamp - 1, load_version, edge_collection=col)
+
+
+def _process_verts(db, vertex_source, timestamp, load_version):
```

Comment:
Would be good to have docstrings for each of these functions

Reply:
What would you recommend for the docstrings? I usually don't add docstrings to internal functions because the string winds up being either huge and recapitulates the function or is pointless like

```
This function processes vertices and updates the database to reflect the current structure of the graph.
```

Reply:
To me, "process_verts" doesn't say much. Something like:
```
"""
For each vertex we're importing, either replace and expire an existing vertex, create a new vertex, or leave an existing vertex unchanged, updating its version.
"""
```